SYCL/Basic/device-selectors-exception.cpp

```diff
@@ -0,0 +1,48 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN:  %t.out
+
+#include <sycl/sycl.hpp>
+using namespace sycl;
+
+#ifdef _WIN32
+#define setenv(name, value, overwrite) _putenv_s(name, value)
+#endif
+
+int refuse_any_device_f(const device &d) { return -1; }
+
+int main() {
+
+  // Disable all available devices
+  setenv("SYCL_DEVICE_FILTER", "", 0);
+
+  // Check exception message for custom device selector
+  try {
+    queue custom_queue(refuse_any_device_f);
+  } catch (exception &e) {
+    assert(strstr(e.what(), "info::device::") == nullptr &&
+           "Incorrect device type in exception message for custom selector.");
+  }
+
+  // Check exception message for pre-defined devices
+  try {
+    queue gpu_queue(gpu_selector_v);
+  } catch (exception &e) {
+    assert(strstr(e.what(), "info::device::gpu") != nullptr &&
```

Comment:
This works too. 
It is though less typical than what other tests do (use std::string and find()), e.g.https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Basic/range_offset_fit_in_int.cpp#L17
or https://github.com/intel/llvm-test-suite/blob/intel/SYCL/Basic/context_platforms.cpp#L26

Reply:
Thank you! Updated.